bin/lib/start.js

```diff
@@ -38,6 +38,25 @@ module.exports = function (program, server) {
         })
       }
 
+      // Read in ENV variables
+      options
+        .map(option => {
+          const configName = option.name.replace(/-([a-z])/g, (_, letter) => letter.toUpperCase())
+          const snakeCaseName = configName.replace(/([A-Z])/g, '_$1')
+          const envName = `SOLID_${snakeCaseName.toUpperCase()}`
+
+          return { configName, envName }
+        })
+        .forEach(({ configName, envName }) => {
+          if (process.env[envName]) {
+            try {
+              argv[configName] = JSON.parse(process.env[envName])
+            } catch (e) {
+              argv[configName] = process.env[envName]
+            }
```

Comment:
This implies that, if a parsing error occurs, we just set the malformed JSON without error. That can't be good.

Let's check if it is JSON indeed (braces?) and only then parse, and not eat errors.

That said, can we avoid JSON at all?

Reply:
The case is actually a little different.

The `JSON.parse` allows `SOLID_SOME_VAR=true` to get a boolean `true` or `SOLID_PORT=1234` to get a numeric `1234` (and not a string of `"1234"`). It also does allow a user to specify `SOLID_SOME_VAR='{"key": "value"}'` to set an object, which I'm not sure is needed, but the option is available.

However if you try set `SOLID_DB_PATH=./my.db`, the string `
./my.db"` is not valid json, which causes the `JSON.parse` to throw, and so in that case we set this value to the string as it was passed in.

I'd be happy to make any changes, or add any documentation you'd like to see.

Reply:
```JavaScript
const raw = process.env[envName]
argv[configName] = /^(d+|true|false)$/.test(raw) ? JSON.parse(raw) : raw
```

Reply:
But actually, the port should be a string, so

```JavaScript
const raw = process.env[envName]
argv[configName] = /^(true|false)$/.test(raw) ? raw === 'true' : raw
```